newIDE/app/src/Leaderboard/LeaderboardProvider.js

```diff
@@ -122,13 +125,30 @@ const reducer = (state: ReducerState, action: ReducerAction): ReducerState => {
         displayOnlyBestEntry: action.payload,
       };
     case 'UPDATE_OR_CREATE_LEADERBOARD':
+      let leaderboardsByIdsWithUpdatedPrimaryFlags;
```

Comment:
you can probably initiate this variable to `const leaderboardsByIdsWithUpdatedPrimaryFlags = {}` so that you just assign value/key afterwards instead of reassigning the object